src/main/java/repo/ReservationRepo.java

```diff
@@ -30,4 +31,14 @@ public void addReservation(Reservation reservation) {
         return result;
     }
 
+    public boolean checkBookReservation(Book book) {
```

Comment:
The name of this function does not tell me what it does.
Maybe renaming to `reservationExists`.

Functions that return boolean values like `true` or `false` are often named like if you would be asking a question.